src/search.rs

```diff
@@ -0,0 +1,147 @@
+use crate::document::advanced::DocumentEncryptUnmanagedResult;
+use crate::document::advanced::*;
+use crate::document::DocumentEncryptOpts;
+use crate::internal::{take_lock, IronOxideErr};
+use crate::{GroupId, IronOxide};
+use async_trait::async_trait;
+use rand::{self, RngCore};
+use std::convert::{TryFrom, TryInto};
+use std::ops::DerefMut;
+
+use ironcore_search_helpers::generate_hashes_for_string;
+
+type Result<A> = std::result::Result<A, IronOxideErr>;
+///The required length of the salt.
+const REQUIRED_LEN: usize = 32;
+
+///The result of creating a new index as well as initializing an IronSimpleSdk.
+///If you only want to create the index, see create_index.
+pub struct CreatedIndexResult {
```

Comment:
Suggest: `CreateIndexResult` or `SearchIndexCreateResult` or `IndexCreateResult`

Reply:
Of the suggestions, I would choose `IndexCreateResult`

Reply:
Shouldn't `create_index` return an `index`?

Reply:
What do you think about putting the encrypted_salt in the BlindIndexSearchSdk struct and getting rid of this altogether?

Maybe this would look like getting rid of `create_index` (or not) and changing `create_index_and_initialize_search` to also return `Result<BlindIndexSearchSdk>`

Reply:
Discussed offline that this type of construction doesn't make sense as the existence of a `BlindIndexSearch` doesn't necessarily mean that an `EncryptedBlindIndexSalt` exists.

So I'm back to suggesting `IndexCreateResult` as a name here, unless someone has something better. I think what bothers me about the current name is that it starts with a verb.